apps/services/user-notification/src/app/modules/notifications/notifications.service.ts

```diff
@@ -151,4 +163,135 @@ export class NotificationsService {
     }
     return template
   }
+
+  async findOne(
+    user: User,
+    id: number,
+    locale: string,
+  ): Promise<RenderedNotificationDto> {
+    // Retrieve the notification
+    const notification = await this.notificationModel.findOne({
+      where: {
+        id: id,
+        recipient: user.nationalId,
+      },
+    })
+
+    if (!notification) {
+      throw new NotFoundException(
+        'Notification not found or does not belong to the user',
+      )
+    }
+
+    // Process and format the notification
+    try {
+      const template = await this.getTemplate(notification.templateId, locale)
+      const formattedTemplate = this.formatArguments(notification, template)
+      return {
+        id: notification.id,
+        messageId: notification.messageId,
+        title: formattedTemplate.notificationTitle,
+        body: formattedTemplate.notificationBody,
+        dataCopy: formattedTemplate.notificationDataCopy,
+        clickAction: formattedTemplate.clickAction,
+        created: notification.created,
+        updated: notification.updated,
+        status: notification.status,
+      }
+    } catch (error) {
+      this.logger.error('Error formatting notification:', error)
+      throw new InternalServerErrorException('Error processing notification')
+    }
+  }
+
+  async findMany(user: User, query: any): Promise<PaginatedNotificationDto> {
+    const templates = await this.getTemplates(query.locale)
+    const paginatedListResponse = await paginate({
+      Model: this.notificationModel,
```

Comment:
Just curious: It's weird to see capitalized properties. Is this interacting with a `C#`?

Reply:
this is required for the library function being used there

Reply:
thats the way that pagination library function wants it .Capital M